cpp/src/lance/io/exec/limit.cc

```diff
@@ -55,14 +56,23 @@ ::arrow::Result<ScanBatch> Limit::Next() {
   auto left = std::max(offset_, seen_);
   auto right = std::min(seen_ + batch_size, offset_ + limit_);
   std::shared_ptr<::arrow::RecordBatch> record_batch;
+  std::shared_ptr<::arrow::Array> indices;
   if (left < right) {
     record_batch = batch.batch->Slice(left - seen_, right - left);
+    if (batch.indices) {
+      indices = batch.indices->Slice(left - seen_, right - left);
```

Comment:
batch is our ScanBatch abstraction right? Would it make sense to create a batch.Slice that returns a structure with batch/indices as appropriate?

Reply:
Fixed.